morpheus/pipeline/pipeline.py

```diff
@@ -232,12 +228,16 @@ def build(self):
 
         logger.info("====Registering Pipeline====")
 
-        self._mrc_executor = mrc.Executor(self._exec_options)
+        exec_options = mrc.Options()
+        exec_options.topology.user_cpuset = f"0-{self._num_threads - 1}"
+        exec_options.engine_factories.default_engine_type = mrc.core.options.EngineType.Thread
+
+        self._mrc_executor = mrc.Executor(exec_options)
 
-        self._mrc_pipeline = mrc.Pipeline()
+        mrc_pipeline = mrc.Pipeline()
 
         def inner_build(builder: mrc.Builder, segment_id: str):
-            logger.info(f"====Building Segment: {segment_id}====")
+            logger.info("====Building Segment: %s ====", segment_id)
```

Comment:
How come you are switching to this format when in most places you are switching to the other?

Reply:
This is a pylint recommendation. For logger calls the string interpolation isn't performed when the configured log-level would cause the call to be suppressed. 

For other strings, pylint recommends the new-ish f-style strings.